<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <script>
        // let promise = new Promise((resolve, reject) => {
        //     let a = 1 + 1;
        //     if (a == 2) {
        //         resolve('Success');
        //     } else {
        //         reject('Failed');
        //     }
        // });

        // promise.then((message) => {
        //     console.log('This is in the then ' + message);
        // }).catch((message) => {
        //     console.log('This is in the catch ' + message);
        // });

        // let myPromise=new Promise((resolve,reject)=>{
        //     let success=true;
        //     if(success){
        //         resolve("Successful");              //prints success if success=true
        //     }
        //     else{
        //         reject("Failed");                   //prints failed if success=false
        //     }
        // });
        // myPromise
        // .then(result=>{
        //     console.log(result);                   
        // });
        // myPromise
        // .catch(error=>{
        //     console.log(error);
        // });

        //chaining promise

//         let promise=new Promise((resolve,reject)=>{
//     resolve(5);  //initial resolve value is 5
// })
// promise
// .then(result=>{
//     console.log(result);
//     return result*2;
// })
// .then(result=>{
//     console.log(result);
//     return result+3  //return a new value
// })
// .then(result=>{
//     console.log(result);
// })
// .catch(error=>{
//     console.log("error",error);
// })

// let promise=new Promise((resolve,reject)=>{
//     resolve(5);

//    })
//    promise
//    .then(setTimeout((result=>{
//        console.log(result);
//        return result*2;
//    },1000)));
//    promise.then(setTimeout((result)=>{               //error why...?
//        console.log(result);
//        result=result*3;
//    },2000));
//    promise.then(result=>{
//        console.log(result);
//    });
//    promise.catch(error=>{
//        console.log("error",error);
//    });

//    let promise = new Promise((resolve, reject) => {
//        resolve(5);
//    });

//    promise
//        .then(result => {
//            return new Promise((resolve) => {
//                setTimeout(() => {
//                    console.log(result);
//                    resolve(result * 2);
//                }, 1000);
//            });
//        })
//        .then(result => {
//            return new Promise((resolve) => {
//                setTimeout(() => {
//                    console.log(result);
//                    resolve(result * 3);
//                }, 2000);
//            });
//        })
//        .then(result => {
//            console.log(result);
//        })
//        .catch(error => {
//            console.log("error", error);
//        });
   
    </script>
</body>
</html>